src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -288,6 +292,14 @@ internal static bool DoSslHandshake(SafeSslHandle context, byte[] recvBuf, int r
                 {
                     sendCount = BioRead(context.OutputBio, sendBuf, sendCount);
                 }
+                catch (Exception e)
+                {
+                    // If we already have handshake exception, ignore any exception from BioRead().
+                    if (handshakeException == null)
+                    {
+                        handshakeException = e;
+                    }
+                }
```

Comment:
Can we change this catch block to something like:

```c#
catch(Exception ex) when(handshakeException != null)
{
   /* do nothing, already throwing another exception */
   NetEventSource.Log("ignoring extra exception", ex);
}
```

This way we log that some other error happened too and also avoid losing stack trace when rethrowing the exception later on.

Reply:
updated. I left out tracing. I want to do tracing update in general as separate pass.